src/tactic/specialize.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2021 Shing Tak Lam. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shing Tak Lam
+-/
+import tactic.doc_commands
+
+/-!
+# `specialize'`
+
+`specialize'` is a variant of specialize, the only difference being the order of the goals generated
+by any underscores used. That is, `specialize'` puts the goals from the underscores first.
+-/
+
+namespace tactic
+namespace interactive
+open interactive interactive.types expr tactic
+
+/--
+A variant of `specialize` that puts the current goal last. That is, if we have a hypothesis
+`h : ∀ x > 0, f x`, then we can use `specialize h x _`, and then we would get two goals. The first
+would be the current goal, the second goal is for the underscore.
+
+With `specialize'`, the goal for any underscores comes first, then the current goal.
+-/
+meta def specialize' (p : parse texpr) : tactic unit :=
+do
+  n ← list.length <$> get_goals,
+  specialize p,
+  n' ← list.length <$> get_goals,
```

Comment:
Could you also do something like:
```
  gs ← get_goals,
  set_goals [],
  specialize p,
  gs' ← get_goals,
  set_goals gs' ++ gs
```

Reply:
No, I don't think `specialize` works when there are no goals.

Reply:
Although this works

```
  g :: gs ← get_goals,
  set_goals [g],
  specialize p,
  g' :: gs' ← get_goals,
  set_goals $ gs' ++ [g'] ++ gs
```